src/main/java/ch/njol/skript/entity/EntityData.java

```diff
@@ -171,21 +179,61 @@ public static void onRegistrationStop() {
 		});
 	}
 
+	/**
+	 * Attempts to get an {@link EntityType} from a {@link Class} extending {@link Entity}.
+	 * Ensures at least one {@link EntityType} can represent an entity class through {@link Class#isAssignableFrom(Class)}.
+	 * @param entityClass The {@link Class} extending {@link Entity}
+	 * @return The exact or assignable {@link EntityType} or {@code null}
+	 */
+	public static <E extends Entity> @Nullable EntityType getEntityType(Class<E> entityClass) {
```

Comment:
```suggestion
	public static <E extends Entity> @Nullable EntityType getBukkitEntityType(Class<E> entityClass) {
```
This should also be in EntityUtils, not EntityData

Reply:
Can't add it in `EntityUtils`. I originally did put it in there, but it was causing an exception to be thrown due to the static field of `EntityUtils` creating `EntityData`s before any/all were registered.
https://github.com/SkriptLang/Skript/blob/1f333d5fc64366282b71f62e92af760a14b3f4d6/src/main/java/ch/njol/skript/entity/EntityData.java#L260-L269

Reply:
Sorry, can you clarify? I don't see how putting it in EntityUtils would cause any entity datas to be created.

Reply:
Whenever the class loader was loading the `EntityData` files, such as `AxolotlData`, which was calling `EntityData#register`.  If I put the new method in `EntityUtils` the static field in `EntityUtils` was creating new  `EntityData`s before all were done being registered. As the stacktrace, the line throwing the exception was the `throw new IllegalStateException` from the previous file lines I sent
https://github.com/SkriptLang/Skript/blob/1f333d5fc64366282b71f62e92af760a14b3f4d6/src/main/java/ch/njol/skript/bukkitutil/EntityUtils.java#L23-L29

Reply:
That's probably because we shouldn't be relying on such flimsy load order to create SPAWNER_TYPES.
You should move that code to a private method and call it when spawner types is needed but not yet created

Reply:
Alrighty